src/Microsoft.AspNet.Mvc.Core/Formatters/InputFormatter.cs

```diff
@@ -107,27 +107,25 @@ public virtual Task<InputFormatterResult> ReadAsync(InputFormatterContext contex
 
         /// <summary>
         /// Returns an <see cref="Encoding"/> based on <paramref name="context"/>'s
-        /// <see cref="MediaTypeHeaderValue.Charset"/>.
+        /// character set.
         /// </summary>
         /// <param name="context">The <see cref="InputFormatterContext"/>.</param>
         /// <returns>
         /// An <see cref="Encoding"/> based on <paramref name="context"/>'s
-        /// <see cref="MediaTypeHeaderValue.Charset"/>. <c>null</c> if no supported encoding was found.
+        /// character set. <c>null</c> if no supported encoding was found.
         /// </returns>
         protected Encoding SelectCharacterEncoding(InputFormatterContext context)
         {
             var request = context.HttpContext.Request;
 
-            MediaTypeHeaderValue contentType;
-            MediaTypeHeaderValue.TryParse(request.ContentType, out contentType);
-            if (contentType != null)
+            if (request.ContentType != null)
             {
-                var charset = contentType.Charset;
-                if (!string.IsNullOrEmpty(charset))
+                var encoding = MediaTypeEncoding.GetEncoding(request.ContentType);
+                if (encoding != null)
                 {
                     foreach (var supportedEncoding in SupportedEncodings)
                     {
-                        if (string.Equals(charset, supportedEncoding.WebName, StringComparison.OrdinalIgnoreCase))
+                        if (encoding.WebName.Equals(supportedEncoding.WebName, StringComparison.OrdinalIgnoreCase))
```

Comment:
Didn't know that was the case, I can change it, even though it's perfectly safe to use in this case (see encoding != null above)